README.md

```diff
@@ -64,137 +87,159 @@ class { 'wsus_client':
 
-##### `target_group`
+If this parameter is set to 'false', Puppet sets `UseWUServer` to false, disabling WSUS updates on the client. If undefined, Puppet does not manage `WUServer` or `UseWUServer`.
 
-The target group that the machine belongs to, note this setting is only respected when allowed from WSUS Server. Valid values are String. Default: 'undef'
+Even if HTTPS is required for authentication, you can use 'http' URLs instead of 'https'. WSUS automatically switches to an HTTPS connection when required and increments the provided port by 1. For example, if the `server_url` is 'http://myserver:8530' and the WSUS server requires HTTPS access, the client automatically uses 'https://myserver:8531' to authenticate, then downloads the updates without encryption via the `server_url`. This performs better than using SSL to encrypt binary downloads.
 
-#### `purge_values`
+**Note**: The `server_url` parameter is central to using wsus_client to manage updates from a WSUS server. While not strictly required to use the class, note that you must manage the `WUServer` and `UseWUServer` registry keys yourself if you do not set `server_url` and [`enable_status_server`][].
 
-Whether to purge the registry values we are not managing under WindowsUpdate parent key. Valid options: 'true' and 'false'. Default: 'false'
+#### `target_group`
 
-## Links
+Sets the client's target group. Valid values: a string. Default: undef.
 
-[Microsoft - Configuring Automatic Updates using the Registry Editor](https://technet.microsoft.com/en-us/library/dd939844(v=ws.10).aspx)
+This [enabled parameter][] is only respected when the WSUS server allows clients to modify this setting via the [`TargetGroup` and `TargetGroupEnabled`][TechNet] registry keys.
+
+### Enabled Parameters
+
+Several `wsus_client` parameters modify two registry keys, one with a value and the other with a Boolean switch. These parameters, known as **enabled parameters**, modify both keys only when assigned a value other than 'false'.
+
+For example, if the [`reschedule_wait_time_minutes`][] parameter takes a valid integer value from '1' to '60', Puppet sets the `RescheduleWaitTimeEnabled` registry key's value to '1' (true). If set to an invalid value or left undefined, Puppet sets the registry key to '0' (false), which disables the feature.
+
+The module's enabled parameters are:
+* [`detection_frequency_hours`][]
+* [`reboot_relaunch_timeout_minutes`][]
+* [`reboot_warning_timeout_minutes`][]
+* [`reschedule_wait_time_minutes`][]
+* [`target_group`][]
 
 ## Limitations
 
-Windows 2003+
+This module requires clients running Windows Server 2003 or newer, and a configured and active [WSUS server][] to use all of the module's options except [`purge_values`][]. For detailed compatibility information, see the [supported module compatibility matrix](https://forge.puppetlabs.com/supported#compat-matrix).
+
+## Development
```

Comment:
One more line you can add at the end of Development:

To see who's already involved, see the list of [contributors](https://github.com/puppetlabs/puppetlabs-wsus_client/graphs/contributors).